src/free-style.ts

```diff
@@ -170,10 +170,11 @@ function stringifyProperties (properties: Properties) {
 
   for (const [name, value] of properties) {
     if (value != null) {
-      if (Array.isArray(value)) {
-        result.push(value.filter(x => x != null).map(x => styleToString(name, x)).join(';'))
+      const value2 = typeof value === 'function' ? value() : value
```

Comment:
The other issue with using functions is that the code is no longer serialisable and there's a minor fear that this will introduce incompatibilities with CSS-in-JS implementation since it can not be statically analysed anymore. For instance, it's possible to use PostCSS with `free-style`, but this breaks that. It may also break other static analysis tooling and infrastructure - vendor prefixing, etc. will all be broken. I'm going to hold off merging this as a result, sorry!